src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketClientFactoryImpl.java

```diff
@@ -173,6 +175,16 @@ public BitbucketUsernameClient getUsernameClient() {
         };
     }
 
+    @Override
+    public BitbucketWebhookSupportedEventsClient getWebhookCapabilities() {
+        return () -> {
+            AtlassianServerCapabilities capabilities = getCapabilityClient().get();
+            String urlStr = capabilities.getCapabilities().get(WEBHOOK_CAPABILITY_KEY);
+            HttpUrl url = parse(urlStr);
```

Comment:
2 potential NPEs here:
* The capability map might not have an entry for `WEBHOOK_CAPABILITY_KEY` in which case `parse` will throw an NPE.
* `parse` can sometimes return `null` but its being passed straight into `makeGetRequest` which assumes the `url` is not null

Reply:
I have added defensive exception throwing. I don't expect supported bitbucket server version to not support webhook.

Reply:
No, but if they try to connect to an unsupported version we want to give them a good error message